_skycap/the-basics/formations-stencils-and-snapshots.md

```diff
@@ -0,0 +1,121 @@
+
+## What is a Stencil?
+
+Stencils are version controlled configuration templates for Kubernetes. They are used together with Formations to provide a secure, easy and flexible way to generating Kubernetes configuration files based on your application requirements.
+
+Stencils support a light-weight [templating language](/skycap/references/stencil_placeholders.html) that is very easy to use while deliberately lacking full scale control flows like programming languages or templating / scripting markup.
+
+Lack of control flows like `if then else` or `for loop` encourages more understandable and maintainable configuration files.
+
+### What are Stencil Groups?
+
+Usually a Formation contains all Stencils you need to deploy an entire application. This approach lets you build an entire stack in a single rendering of your Formation against a Kubernetes cluster. 
+
+However, this might not always be what you need. You might want to render and apply only a subset of the Stencils available in a Formation.  For example when deploying a single service or running a database migration job, you may not want to deploy the entire stack.
+
+StencilGroups help you group the Stencils of a Formation into dynamic or static subsets so they can be rendered together.
+
+## What is a Snapshot?
+
+A Snapshot is the state of your application at a point in time. It includes all your applications images and their unique tags, environment variables and configuration items that are defined in your Skycap application. 
+
+<img src="/assets/skycap/retagging.png" width="600px">
+
+All of these components are then stored in a private repository on your Cloud 66 account so that you can roll back easily at any time.
+
+Snapshots act like a "time machine" for your code and configuration so you can redeploy your entire application as it was at any point in time or quickly revert your Kubernetes configurations if you need to.
+
+This also prevents issues when using the `latest` tag on your Docker images. If a `latest` tag is used in your service definition, a retagging will "freeze" that version at the time the snapshot is taken.
+
+**An infrastructure time machine**
+
+While Snapshots provide a way to "freeze" your code and other application components in time, they also make it possible to re-apply old code to new configuration. This is particularly useful in the following scenarios:
+
+* Rolling out old code a "playground" cluster for diagnostics.
+* Redeploying the same code with different versions of configuration for optimization purposes.
+* Rolling back configuration changes without redeploying the code.
+* Deploying the production version to a local cluster (like minikube) for debugging
```

Comment:
Is it worth defining *minikube* here? Or providing a link?

Reply:
Good idea - have added a link.